schematypes_test.go

```diff
@@ -48,4 +49,14 @@ func TestMessageInterface(t *testing.T) {
 	if seq != 199 {
 		t.Errorf("got seq=%d, want 199", seq)
 	}
+
+	// Test adherence to the ResponseMessage interface.
+	f2 := func(rm ResponseMessage) int {
+		return rm.GetResponse().RequestSeq
```

Comment:
Add also tests for rm.GetSeq() and errorResonseStruct.GetSeq() for full coverage/documentation of what these interfaces and embedding enable?

Reply:
Added test for `rm.GetSeq()`. Testing `errorResonseStruct.GetSeq()` feels too much because the embed path is directly written down in the code. The idea for testing interface implementation is that interfaces are implemented implicitly so this can slip by moving methods around.

Also refactored the test to avoid placing too much in a single `Test*` function. Using a `makeErrorResponse` constructor to prevent global collision with a `errorResponseStruct` already defined globally in another test file.